packages/cli-e2e/utils/login.ts

```diff
@@ -0,0 +1,109 @@
+import {Page} from 'puppeteer-core';
+import type {Browser} from 'puppeteer-core';
+import {ChildProcessWithoutNullStreams, spawn} from 'child_process';
+import {answerPrompt, CLI_EXEC_PATH, isYesNoPrompt} from './cli';
+
+function isLoginPage(page: Page) {
+  return page.url().match(/https:\/\/platform.*cloud\.coveo\.com\/login/);
+}
+
+function isLoggedIn(page: Page) {
+  return page.url().match(/http:\/\/127\.0\.0\.1:32111/);
+}
+
+function waitForLoginPage(browser: Browser) {
+  return new Promise<Page>((resolve) => {
+    browser.on('targetchanged', (page: Page) => {
```

Comment:
Sorry, I'm not sure to understand how I can use `promisify` here :S

Reply:
> Takes a function following the common error-first callback style, i.e. taking a (err, value) => ... callback as the last argument, and returns a version that returns promises.

So, actually never mind, it is not an `an error first callback pattern`, so it would not work in this case.

If it had been however, you could use (I think) `utils.promisify` on `browser.on('targetChanged')`

So you'd get something like: 

```
const page = await utils.promisify(browser.on('targetchanged'))
```

Anyway, does not apply in this case ! NVM me ;)